setuptools/_importlib.py

```diff
@@ -10,3 +10,10 @@
     import importlib_resources as resources  # pragma: no cover
 else:
     import importlib.resources as resources  # noqa: F401
+
+
+if not hasattr(metadata, "EntryPoints"):  # pragma: no cover
+    raise ImportError(
+        "`importlib-metadata` version is incompatible with `setuptools`. "
+        "Please install importlib-metadata>=4.6"  # Equivalent to stdlib's in 3.10
```

Comment:
The choice of `importlib-metadata>=4.6` is based on the fact that the `if` statement above accepts stdlib's version for Python 3.10 and according to [`importlib_metadata`'s docs](https://github.com/python/importlib_metadata) that is equivalent to version 4.6.

So if the codebase works with importlib.metadata in Python 3.10 it should also work with `importlib_metadata` 4.6. I am not sure why `pyproject.toml` requires `importlib_metadata>=6`...